jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/MinimumCostFlowAlgorithm.java

```diff
@@ -30,91 +32,52 @@
  * @param <E> graph edge type
  * @author Timofey Chudakov
  */
-public interface MinimumCostFlowAlgorithm<V, E> {
+public interface MinimumCostFlowAlgorithm<V, E> extends FlowAlgorithm<V,E>{
 
     /**
-     * Calculates feasible flow of minimum cost for the minimum cost flow problem. If minimum cost
-     * flow in not unique, the algorithm chooses the result arbitrarily.
+     * Calculates feasible flow of minimum cost for the minimum cost flow problem.
      *
+     * @param minimumCostFlowProblem minimum cost flow problem
      * @return minimum cost flow
      */
-    MinimumCostFlow<E> getMinimumCostFlow();
+    MinimumCostFlow<E> getMinimumCostFlow(MinimumCostFlowProblem minimumCostFlowProblem);
```

Comment:
Why don't we use the generic version of the `MinimumCostFlowProblem<V, E>`?

Reply:
Oops, somehow missed this. Thanks for catching this!